src/main/java/teammates/common/datatransfer/questions/FeedbackMsqQuestionDetails.java

```diff
@@ -756,4 +953,161 @@ private double divideOrReturnZero(double numerator, int denominator) {
         return (denominator == 0) ? 0 : numerator / denominator;
     }
 
+    /**
+     * Calculate the Per Recipient Statistics for MSQ questions.
+     */
+    private static class PerRecipientStatistics {
+        List<String> choices;
+        boolean otherEnabled;
+        ResponseStatistics responseStats;
+
+        PerRecipientStatistics(FeedbackMsqQuestionDetails questionDetails,
+                ResponseStatistics responseStats) {
+            this.choices = questionDetails.getMsqChoices();
+            this.otherEnabled = questionDetails.getOtherEnabled();
+            this.responseStats = responseStats;
+        }
+
+        /**
+         * Calculates the number of responses for each response attribute.
+         */
+        private Map<String, Integer> getNumberOfResponsesForEachResponse(FeedbackMsqResponseDetails responseDetails,
```

Comment:
Similar code in `getNumberOfResponses`?

Reply:
Actually, a more fancy way to do the stats class would be:

- We have basic abstract class named `MultipleOptionStatistics` (can put in `MCQQuestionDetails`) (May be a better name). There are some common methods like `calculateWeightedPercentagePerOption` and `getPerRecipientResponseStatsHeaderCsv`
- `MCQStatistics` extends `MultipleOptionStatistics` and have some specific methods
- `MSQStatistics` extends `MultipleOptionStatistics` and have some specific methods (like `getNumberOfResponsesForEachResponse `)

Reply:
I really like this idea. I should have thought about it earlier. :man_facepalming: